packages/@lwc/template-compiler/src/shared/types.ts

```diff
@@ -8,6 +8,7 @@ import { CompilerDiagnostic } from '@lwc/errors';
 
 export type TemplateCompileResult = {
```

Comment:
To indicate that TemplateParseResult is a subset of TemplateCompileResult.
```suggestion
export type TemplateParseResult = {
    root?: Root;
    warnings: CompilerDiagnostic[];
};

export type TemplateCompileResult = {
    code: string;
} & TemplateParseResult;
```